src/librustc/mir/repr.rs

```diff
@@ -32,6 +32,10 @@ pub struct Mir<'tcx> {
     /// that indexes into this vector.
     pub basic_blocks: Vec<BasicBlockData<'tcx>>,
 
+    /// List of lexical scopes; these are referenced by statements and
+    /// used (eventually) for debuginfo. Indexed by a `ScopeId`.
+    pub scopes: ScopeDataVec,
```

Comment:
@nagisa 

> I feel like it might be a good idea to have spans in the side table, along with scopes.

While every reference to a scope has a span, those spans are different at each point, so we can't put them in the same table. It might be useful to put them in different tables. But then I think MIR would be the wrong place to do this -- more like codemap. (And, to some extent, it already does.)